### ISSUE TITLE
mupen64plus.sh issue with TexturePack

### ISSUE THREAD

**Issue by MNieddu91** [gonetz/GLideN64#1543]
Hello,
after several days of trying to get a texture pack to work under GLideN64 i managed it by fixed it.
If `EnableFBEmulation = True` in `mupen64plus.cfg` texture packs are not loading properly in GLideN64 and it seems the resolution ist destroyed as well. Screen seems very creepy then.
the Problem is, that EnableFBEmulation is set to True by default in line 208 of `mupen64plus.sh`.
Probably we can fix the issue, by checking if there is a texture pack for current started ROM and if there is one set `EnableFBEmulation = False`
Is it only me having the problem? We need a workaround or fix for that ?

PS.: I'm using RetroPie 4.2 on a RPi3, but it seems not to be an RetroPie issue.

Greets Marco

**Comment by AmbientMalice** [comment]
What texture pack are you trying to use? Framebuffer emulation is generally essential. But Raspberry Pis are kinda buggy, and compatibility is a problem. Someone more familiar with the hardware can probably help. Maybe @loganmc10 has some idea?

**Comment by loganmc10** [comment]
Yeah if I knew the texture pack I could test it. The Raspberry Pi only supports OpenGL ES2, and it has some issues with Framebuffer Emulation to begin with, so there's a good chance the issue is specific to the Raspberry Pi, or it could also be something the RetroPie script that launches mupen64plus is doing

**Comment by MNieddu91** [comment]
This 'Bug' is a bit more difficult.
First of all it seems like every Texture Pack is nor working if `EnableFBEmulation = True`, but it's also a kind of resolution issue.
Following screenshot shows Mario Kart with Texture Pack and `EnableFBEmulation = True` (mupen64plus-GLideN64-highres):
[http://imgur.com/c5Z7Hln](http://imgur.com/c5Z7Hln)
[http://imgur.com/zXLFb1m](http://imgur.com/zXLFb1m)

Here you can see it with `EnableFBEmulation = False` (mupen64plus-GLideN64-highres) with my workaround (described below):
[http://imgur.com/zWjcxnu](http://imgur.com/zWjcxnu)
[http://imgur.com/CT02k1A](http://imgur.com/CT02k1A)

I managed this, with a for me working workaround, because i only have these texture Packs, but it's not a good solution over all. But working for me at the moment.
I added some Code in the `useTexturePacks()` function to check if i start a game for which i installed a texture pack, if yes set `EnableFBEmulation = False`. 
Code i'm using:
```
   # these games don't load HighRes if EnableFBEmulation is enabled
    local highres_blacklist=(
		"Mario Kart 64 (Hacked)"
		"Mario Kart 64 (USA)(Fixed)"
		"Paper Mario (USA)"
		"Super Mario 64 Multiplayer"
		"Super Smash Bros. (USA)"
		"Super Mario 64 (USA)"
		"Legend of Zelda, The - Ocarina of Time (USA) (Rev B)"
		"Legend of Zelda, The - Majora's Mask (USA)"
    )

	for game in "${highres_blacklist[@]}"; do
		if [[ "$ROM" == *"$game"* ]]; then
			iniSet "EnableFBEmulation" "False"
		fi
        done
```

I tried it with several texture packs, it's all the same for each of them. I tried Kerber 2K for Mario Kart, Mollymutt for all kinds of Super Mario 64 and Paper Mario and Djipis Cell Shade for Zelda.

The second thing is, that ALL EUROPE Games i have no texture pack for there is another issue which also has to do with 'EnableFBEmulation'. But it seems to be only come up with europe version of Roms.
All described europe games shown up as in the screenshot below (picture not centred and cut off) if `EnableFBEmulation = False`. So i need to set `EnableFBEmulation = True` to set the 'screen' to right position.

Screenshot Game without texture pack `EnableFBEmulation = False`:
[http://imgur.com/7jj1idf](http://imgur.com/7jj1idf)

Screenshot Game without texture pack `EnableFBEmulation = True`:
[http://imgur.com/8vtbOeX](http://imgur.com/8vtbOeX)

After all I think it's not just a texture pack issue, maybe more with starting in wrong resolution or why seems the game so creepy ? 

If you need further Information about my system feel free to contact me, but please tell me where to get these informations. Maybe some kind of Command which lists up all information?

Greets Marco

**Comment by loganmc10** [comment]
I'll look into the texture pack issue in a bit, but the issue with European ROM's has actually been reported before I realized:

https://github.com/gonetz/GLideN64/issues/1428

I thought that it was Raspberry Pi specific, but I'm assuming @gizmo98 was using E roms in that test.

@gonetz , it turns out that since the VI refactoring, European ROM's aren't displaying correctly with Framebuffer Emulation disabled, USA ROM's work fine.

**Comment by gizmo98** [comment]
I would be really happy if we could fix this European roms problem. Some gsmes run much faster on a pi if fb rmulation is disabled.

**Comment by MNieddu91** [comment]
One idea: 
Maybe best is to check which folders exists in `local/share/mupen64plus/hires_texture` and compare to ROM header name?! But i have no idea yet how to get ROM Header name. 

Nice weekend everyone

**Comment by gizmo98** [comment]
@gonetz 
Do you have an idea why european roms are broken if framebuffer emulation is disabled?

**Comment by gonetz** [comment]
**Broken** == cut or not centered? 
Video Interface emulaton is required to properly place frame on screen. VI disabled when fb emulation is off.

**Comment by gizmo98** [comment]
Not centered. It worked last year so it is a regression. 
https://github.com/gonetz/GLideN64/issues/1428#issue-215223901
